src/main/java/com/entropyteam/entropay/employees/services/ContractService.java

```diff
@@ -28,14 +33,45 @@ public class ContractService extends BaseService<Contract, ContractDto, UUID> {
 
     @Autowired
     public ContractService(ContractRepository contractRepository, CompanyRepository companyRepository,
-            EmployeeRepository employeeRepository, RoleRepository roleRepository,
-            SeniorityRepository seniorityRepository) {
+                           EmployeeRepository employeeRepository, RoleRepository roleRepository,
+                           SeniorityRepository seniorityRepository) {
         this.contractRepository = contractRepository;
         this.companyRepository = companyRepository;
         this.employeeRepository = employeeRepository;
         this.roleRepository = roleRepository;
         this.seniorityRepository = seniorityRepository;
+    }
+
+    @Transactional
+    @Override
+    public ContractDto create(ContractDto contractDto) {
+        contractRepository.findContractByEmployeeAndActiveIsTrue(
+                        employeeRepository.findById(contractDto.employeeId()).orElseThrow()
+                )
+                .ifPresent(existent -> {
+                    existent.setActive(false);
+                    contractRepository.saveAndFlush(existent);
```

Comment:
Did you set the contract end date after set it to inactive?

Reply:
Nope! I forgot. 😃